crates/swc_ecma_minifier/src/analyzer/mod.rs

```diff
@@ -1070,15 +1088,25 @@ where
     }
 
     #[cfg_attr(feature = "debug", tracing::instrument(skip(self, n)))]
-    fn visit_switch_case(&mut self, n: &SwitchCase) {
-        n.test.visit_with(self);
+    fn visit_switch_stmt(&mut self, n: &SwitchStmt) {
+        n.discriminant.visit_with(self);
 
-        {
+        let mut fallthrough = false;
+
+        for case in n.cases.iter() {
             let ctx = Ctx {
                 in_cond: true,
                 ..self.ctx
             };
-            self.with_ctx(ctx).visit_in_cond(&n.cons);
+            if fallthrough {
+                self.with_child(case.span.ctxt, ScopeKind::Block, |child| {
```

Comment:
Does this work? I think we should pach `unique_scopes` too

Reply:
This is mainly for clearing `initialized_vars`. In that case it works